src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,56 @@
 package core.basesyntax;
 
-class Pair {
+class Pair<F, S> {
+    private F first;
+    private S second;
 
+    private Pair(F first, S second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    public static <F, S> Pair of(F first, S second) {
+        return new Pair(first, second);
+    }
+
+    public F getFirst() {
+        return first;
+    }
+
+    public S getSecond() {
+        return second;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = 0;
+        result += getFirst() == null ? 0 : 31 * getFirst().hashCode();
+        result *= getSecond() == null ? 1 : 31 * getSecond().hashCode();
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object pair) {
+        if (pair == this) {
+            return true;
+        }
+        if (pair == null) {
+            return false;
+        }
+        if (pair.getClass().equals(Pair.class)) {
+            Pair<F, S> anotherPair = (Pair<F, S>) pair;
+            return anotherPair.first != null && anotherPair.first.equals(first)
+                    && anotherPair.second != null && anotherPair.second.equals(second)
+                    || anotherPair.first == null && first == null
+                    || anotherPair.second == null && second == null;
```

Comment:
this is not quite correct

let's recall the flow that we did for comparing `rectangle` fields in the previous task

Reply:
Sure, I recently submitted the right solution for the rectangle and got approved.